### PR TITLE
feature/107-rfc-for-page-content-resolving-endpoint

### PR DESCRIPTION
**Pull request by rmakara**
Purpose of that pull requests is to create an environment for code based discussions on Page Content Resolving Endpoint.

Please see files changed.

**Issue:**
#107

### CODE REVIEW

File: rfc/Resolver/PageContentResolver.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  pageType: string;
+  pathInfo: string;
+  seoPathInfo: string;
+  isCanonical: boolean;
+  resourceIdentifier: string;
+  data: CategoryPageHeadless | ProductPageHeadless;
+}
+
+interface CategoryPageHeadless {
+  CategoryCmsPage: CmsPage;
```
Comment (patzick) [inline rfc/Resolver/PageContentResolver.ts]:
```suggestion
  cmsPage: CmsPage;
```

File: rfc/Resolver/PageContentResolver.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  availableFilters: AvailableFilters;
+  availableSortings: AvailableSortings;
+  categories: CategoryForHeadless[];
+}
+
+interface ProductPageHeadless {
+  ProductCmsPage: CmsPage;
```
Comment (patzick) [inline rfc/Resolver/PageContentResolver.ts]:
```suggestion
  cmsPage: CmsPage;
```

File: rfc/ResolverInterface.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+enum PageType {
+  category = "category",
+  product = "product",
+  cmsPage = "cms-page"
+}
+
+export interface PageCriteria extends SearchCriteria {
+  path: string; // path + other possible criteria
+  includes?: Array<string>; // array of fields to be included (removes others from payload)
```
Comment (patzick) [inline rfc/ResolverInterface.ts]:
```suggestion
  includes?: string[]; // array of fields to be included (removes others from payload)
```

File: packages/shopware-6-client/src/interfaces/models/content/cms/CmsPage.ts [hunk 4]
```diff
@@ -1 +1,119 @@
-export interface CmsPage {}
+import { CustomField } from "../../common/CustomField";
```
[the remark this answers is not in the document]
Comment (mkucmus) [reply packages/shopware-6-client/src/interfaces/models/content/cms/CmsPage.ts]: it's something different, but looks similar

File: packages/shopware-6-client/src/interfaces/models/content/cms/CmsPage.ts [hunk 5]
```diff
@@ -1 +1,119 @@
...
+  landingpage = "landingpage"
+}
+export interface CmsPage {
+  type: CmsPageType;
+  name: string;
+  customFields: [CustomField] | null;
+  locked: boolean;
+  _uniqueIdentifier: string;
+  versionId: string | null;
+  translated: any;
```
[the remark this answers is not in the document]
Comment (mkucmus) [reply packages/shopware-6-client/src/interfaces/models/content/cms/CmsPage.ts]: yes, as long as we don't know what exactly can be in here

**Pull request merged**